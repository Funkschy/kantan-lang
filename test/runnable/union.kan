import "io";

type Test union {
    i: i8,
    s: string
}

type Same union {
    i: i32,
    j: i32
}

def main() {
    let t = Test { i: 42 };
    io.printf("%d\n", t.i);
    t = Test { s: "hello" };
    io.printf("%s\n", t.s);

    t.i = 33;
    io.printf("%d\n", t.i);

    let p = &t.i;
    *p = 2;
    io.printf("%d\n", *p);

    let same = Same { i: 42 };
    io.printf("%d\n", same.j);
}

