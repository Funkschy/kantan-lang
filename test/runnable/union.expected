42
hello
33
2
42
